src/LostDogForm.js

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import AnnouncementData from "./AnnouncementData";
+import {connect} from "react-redux";
+
+class LostDogForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.lostDogRef = React.createRef();
+        this.state = {
+            name: '',
+            dogsname: '',
+            dogsbread: '',
+            location: '',
+            message: '',
+            phoneNumber: '',
+            email: '',
+            filename: '',
+            imgSource:'',
+            uploading: false
+        };
+
+        this.handleInputChange = this.handleInputChange.bind(this);
+    }
+
+
+    // https://stackoverflow.com/questions/9267899/arraybuffer-to-base64-encoded-string
+
+    arrayBufferToBase64( buffer ) {
+        var binary = '';
+        var bytes = new Uint8Array( buffer );
+        var len = bytes.byteLength;
+        for (var i = 0; i < len; i++) {
+            binary += String.fromCharCode( bytes[ i ] );
+        }
+        return window.btoa( binary );
+    }
+
+
+    handleInputChange(event) {
+
+
+        // console.log(event)
+        const target = event.target;
+        const name = target.name;
+        const size = target.size;
+
+
+        this.setState({
+            [event.target.name]: event.target.value
+        });
+        console.log(name)
+        console.log(target.files)
+        console.log(target)
+    }
+
+    setThumbnail(base64String){
+        this.setState({imgSource:"data:image/png;base64," + base64String})
+    }
+
+    handleFileChange(event) {
+        const reader = new FileReader();
+        reader.onload = () => {
+            const base64 = this.arrayBufferToBase64(reader.result);
+            this.setThumbnail(base64)
+
+        };
+        reader.onerror = (error)=>{
+            console.log(error);
+        };
+        reader.readAsArrayBuffer(event.target.files[0]);
+    }
+
+
+    handleSubmit(e) {
+        e.preventDefault();
+        this.setState({uploading:true})
+        const an = new  AnnouncementData();
```

Comment:
why not just call variable `announcementData` it would make code so much more readable